maintenance/Maintenance.php

```diff
@@ -1112,6 +1112,48 @@ public function setDB( &$db ) {
 	}
 
 	/**
+	 * Begin a transcation on a DB
+	 *
+	 * This method makes it clear that begin() is called from a maintenance script,
+	 * which has outermost scope. This is safe, unlike $dbw->begin() called in other places.
+	 *
+	 * @param DatabaseBase $dbw
+	 * @param string $fname Caller name
+	 * @since 1.27
+	 */
+	protected function beginTransaction( DatabaseBase $dbw, string $fname ) {
+		$dbw->begin( $fname );
+	}
+
+	/**
+	 * Commit the transcation on a DB handle and wait for replica DBs to catch up
+	 *
+	 * This method makes it clear that commit() is called from a maintenance script,
+	 * which has outermost scope. This is safe, unlike $dbw->commit() called in other places.
+	 *
+	 * @param DatabaseBase $dbw
+	 * @param string $fname Caller name
+	 * @since 1.27
+	 */
+	protected function commitTransaction( DatabaseBase $dbw, string $fname ) {
+		$dbw->commit( $fname );
+	}
+
+	/**
+	 * Rollback the transcation on a DB handle
+	 *
+	 * This method makes it clear that rollback() is called from a maintenance script,
+	 * which has outermost scope. This is safe, unlike $dbw->rollback() called in other places.
+	 *
+	 * @param DatabaseBase $dbw
+	 * @param string $fname Caller name
+	 * @since 1.27
+	 */
+	protected function rollbackTransaction( DatabaseBase $dbw, string $fname ) {
```

Comment:
What exactly is the point of having methods like `beginTransaction()`, `commitTransaction()` and `rollbackTransaction()` that are nothing more than `function add($a, $b) { return $a + $b; }`. Why cannot we simply call `$dbw->commit( $name );` where we need, just as we would use `$a + $b` rather than introduce a special function for that? :)